### PR TITLE
[BEAM-8911] New non-vendored Guava version: 25.1-jre

### PR DESCRIPTION
**Pull request by suztomo**
https://issues.apache.org/jira/browse/BEAM-8911

Now unused non-vendored Guava dependencies are removed ([BEAM-8747](https://issues.apache.org/jira/browse/BEAM-8747)) and Hadoop client version is upgraded to 2.8.5 from 2.7 ([BEAM-8822](https://issues.apache.org/jira/browse/BEAM-8822)). It's time to upgrade Guava.

# New Guava version: 25.1-jre
CharMatcher.DIGIT, used by org.apache.cassandra.io.sstable.format.SSTableFormat, is removed since Guava 26.0. So this PR is setting Guava 25.1:

https://guava.dev/releases/25.1-jre/api/docs/com/google/common/base/CharMatcher.html
https://guava.dev/releases/26.0-jre/api/docs/com/google/common/base/CharMatcher.html

Guava has 2 variations "-jre" for Java8 (stream/lambda support) and "-android" for Java7. Gcsio:1.9.16 needs ImmutableSet.toImmutableSet() which is available in Guava "-jre". [detail](https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1028#issuecomment-55768092)

So this PR is setting Guava version "25.1-jre".

# Which modules use non-vendored Guava?

```
suztomo@suxtomo24:~/beam6$ find . -name build.gradle | xargs grep library.java.guava
./sdks/java/core/build.gradle:  shadowTest library.java.guava_testlib
./sdks/java/io/google-cloud-platform/build.gradle:  compile library.java.guava
./sdks/java/io/kinesis/build.gradle:  compile library.java.guava
./sdks/java/io/kinesis/build.gradle:  testCompile library.java.guava_testlib
./sdks/java/extensions/sql/zetasql/build.gradle:  compile library.java.guava
```

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -372,7 +372,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_cloud_core_version = "1.61.0"
     def google_cloud_spanner_version = "1.6.0"
     def grpc_version = "1.17.1"
-    def guava_version = "20.0"
+    def guava_version = "25.1-jre"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Cassandra-all version 3 does not work with Guava >= 26 ([detail](https://issues.apache.org/jira/browse/BEAM-8911))

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
java.lang.RuntimeException: Workflow failed. Causes: Project apache-beam-testing has insufficient quota(s) to execute this workflow with 1 instances in region us-central1. Quota summary (required/available): 1/11893 instances, 1/0 CPUs, 250/297535 disk GB, 0/3647 SSD disk GB, 1/68 instance groups, 1/69 managed instance groups, 1/310 instance templates, 1/599 in-use IP addresses.

https://builds.apache.org/job/beam_PostCommit_Java_PR/265/testReport/junit/org.apache.beam.examples.complete/TopWikipediaSessionsIT/testE2ETopWikiPages/

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Python PreCommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Python PreCommit

**Comment by suztomo** [comment]
R: @kennknowles, @iemejia 

Java SDK Post Commit Tests succeeded.

**Comment by iemejia** [comment]
Run Python PreCommit

**Comment by lgajowy** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by lgajowy** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by iemejia** [comment]
Run Dataflow ValidatesRunner

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
```
10:11:23     SEVERE: 2019-12-09T15:11:22.086Z: An OutOfMemoryException occurred. Consider specifying higher memory instances in PipelineOptions.
10:11:23     java.lang.RuntimeException: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.ExecutionError: java.lang.OutOfMemoryError: Java heap space
10:11:23     	at org.apache.beam.runners.dataflow.worker.util.common.worker.BatchingShuffleEntryReader$ShuffleReadIterator.fillEntries(BatchingShuffleEntryReader.java:132)
```
https://builds.apache.org/job/beam_BiqQueryIO_Streaming_Performance_Test_Java/235/console

**Comment by suztomo** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by lgajowy** [comment]
Run Dataflow ValidatesRunner

**Comment by kennknowles** [comment]
run sql postcommit

**Comment by suztomo** [comment]
Flink Runner Nexmark Tests

**Comment by suztomo** [comment]
Spark Runner Nexmark Tests

**Comment by suztomo** [comment]
Run Java Flink PortableValidatesRunner Batch

**Comment by suztomo** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Comment by suztomo** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by suztomo** [comment]
Run Java PortabilityApi PostCommit

**Comment by suztomo** [comment]
Run Java examples on Dataflow with Java 11

**Comment by suztomo** [comment]
Run Java Portability examples on Dataflow with Java 11

**Comment by suztomo** [comment]
Run Apex ValidatesRunner

**Comment by suztomo** [comment]
Run Flink ValidatesRunner

**Comment by suztomo** [comment]
Run Gearpump ValidatesRunner

**Comment by suztomo** [comment]
Run Dataflow PortabilityApi ValidatesRunner

**Comment by suztomo** [comment]
Run Samza ValidatesRunner

**Comment by suztomo** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Dataflow Runner Nexmark Tests

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Dataflow Runner Nexmark Tests

**Comment by suztomo** [comment]
@kennknowles Now they're green; I see some of them didn't trigger builds. Would you merge this?

**Comment by suztomo** [comment]
R: @lgajowy 
Found your name in [BEAM-5559](https://issues.apache.org/jira/browse/BEAM-5559)).

**Comment by suztomo** [comment]
Kenn, thanks!

**Comment by lgajowy** [comment]
Great that all the checks passed. Thank you for doing this @suztomo :)

**Pull request merged**